### ISSUE TITLE
Investigate LLVM integration via llvm-primitives-alloc

### ISSUE THREAD

**Issue by waywardmonkeys** [dylan-lang/opendylan#1049]
Instead of having some small C stubs in ``mps-collector.c``, we could consider just having ``llvm-primitives-alloc.dylan`` go ahead and emit direct calls to MPS.

@housel?

(and @rptb1, you might find this interesting.)

**Comment by rptb1** [comment]
Even better if you could emit the LLIR for in-line allocation, and at least partially deal with #235.

**Comment by housel** [comment]
Yes, obviously this could be done, and indeed it seems that it was always the intention of the MPS design that it be done this way, in order that the initial allocation request, object initialization, and commit can be separated. (At present primitive_copy_r() has to clear slots in allocated closure objects s and then copy closed-over values after the commit). I think having some way of passing code generation options into the back-end should be a precondition, however, so that the other two allocators can continue to work with the current interface.
